src/Files.App/Data/Commands/Manager/CommandManager.cs

```diff
@@ -14,8 +14,8 @@ internal sealed class CommandManager : ICommandManager
 	{
 		private readonly IGeneralSettingsService settings = Ioc.Default.GetRequiredService<IGeneralSettingsService>();
 
-		private readonly IImmutableDictionary<CommandCodes, IRichCommand> commands;
-		private IImmutableDictionary<HotKey, IRichCommand> hotKeys = new Dictionary<HotKey, IRichCommand>().ToImmutableDictionary();
+		private readonly ImmutableDictionary<CommandCodes, IRichCommand> commands;
```

Comment:
We can change it to `FrozenDictionary`.

Reply:
Agree and tried to change, but the interface requires this method:

![image](https://github.com/files-community/Files/assets/52143624/de5b8358-3579-42e1-9bf2-0602291cded5)

Reply:
![image](https://github.com/files-community/Files/assets/52143624/a185b348-ab74-4004-83e0-af82322e44f2)

Removing this unnecessary implementation, we can use `FrozenDictionary`, and I think it would be beneficial to increase read speed of this often read dictionary.

Reply:
This should work.
```csharp
public IEnumerator<IRichCommand> GetEnumerator() => (commands.Values as IEnumerable<IRichCommand>).GetEnumerator()
```

Reply:
> This should work.
> 
> ```cs
> public IEnumerator<IRichCommand> GetEnumerator() => (commands.Values as IEnumerable<IRichCommand>).GetEnumerator()
> ```

Nice it worked, should we also use `FrozenDictionary` at `ModifiableCommandManager.cs`?

Reply:
> should we also use FrozenDictionary at ModifiableCommandManager.cs?

Good point! We should.

Reply:
Done!

Reply:
I think hotKeys should remain ImmutableDictionary because it could be reinitialized.